README.md

```diff
@@ -10,12 +10,93 @@ Basic interface for manifolds in Julia.
 The project [`Manifolds.jl`](https://github.com/JuliaManifolds/Manifolds.jl)
 is based on this interface and provides a variety of manifolds.
 
-## Number system
+## Main types
 
-A number system represents the field a manifold is based upon.
-Most prominently, these are real-valued (`ℝ`) and complex valued (`ℂ`) fields that
-parametrize certain manifolds.
-A further type to represent the field of quaternions (`ℍ`) can also be used.
+This packages provides an  [`AbstractManifold`](https://juliamanifolds.github.io/Manifolds.jl/stable/interface.html#ManifoldsBase.AbstractManifold) type, that is parametrized by its field, i.e., whether elements on the manifold are real (`ℝ`) or complex (`ℂ`) or even quaternions (`ℍ`).
+A manifold is a set of points. These are usually represented by (real-, complex-, quarternion-) valued arrays, most prominently matrices.
+Sometimes they are also vectors or tensors.
+For more advanced points, you can also use subtypes [`AbstractManifoldPoint`](https://juliamanifolds.github.io/Manifolds.jl/latest/interface.html#ManifoldsBase.AbstractManifoldPoint).
+Sometimes points on a manifold can be represented in different forms.
+While a first (default) representation can use just arrays, any further representation _must_ use a subtype to dispatch on. Then the default case can (and should) also have a default subtype that just unwrapps to the internal vallue (see [`@default_manifold_fallbacks`]()).
+
+## Interface design
+
+The interface for a manifold is defined to be as generic as possible, such that applications can be implemented as independent as possible from an actual manifold.
+This way, algorithms like those from [`Manopt.jl`](https://manoptjl.org) can be implemented on _arbitrary_ manifolds.
+
+The main design criteria for the interface are
+
+* aims to also provide _efficient_ and _inplace_ computations whenever possible.
+* provide a high level interface that is easy to use
+
+Therefore this interface has 3 main features, that we will explain using two (related)
+concepts, the [exponential map](https://en.wikipedia.org/wiki/Exponential_map_(Riemannian_geometry)) that maps a tangent vector ``X`` at a point ``p`` to a point ``p`` or mathematically ``\exp_p;:T_p\mathcal M \to \mathcal M`` and its generalisation, a [retraction]() ``\operatorname{retr}_p`` with same domain and range.
```

Comment:
```suggestion
concepts, the [exponential map](https://en.wikipedia.org/wiki/Exponential_map_(Riemannian_geometry)) that maps a tangent vector ``X`` at a point ``p`` to a point ``q`` or mathematically ``\exp_p:T_p\mathcal M \to \mathcal M`` and its generalisation, a [retraction]() ``\operatorname{retr}_p`` with same domain and range.
```
Usually it maps to a different point :wink: .

Reply:
Ah, well seen, yes that of course should have been a q.